Extensions/TileMap/tilemapcollisionmaskruntimeobject.ts

```diff
@@ -0,0 +1,490 @@
+      ) {
+        this.setTilesetJsonFile(newObjectData.content.tilesetJsonFile);
+      }
+      if (oldObjectData.content.debugMode !== newObjectData.content.debugMode) {
+        this.setDebugMode(newObjectData.content.debugMode);
+      }
+      if (oldObjectData.content.fillColor !== newObjectData.content.fillColor) {
+        this.setFillColor(
+          this.rgbToNumber(
+            gdjs.rgbOrHexToRGBColor(newObjectData.content.fillColor)
+          )
+        );
+      }
+      if (
+        oldObjectData.content.outlineColor !==
+        newObjectData.content.outlineColor
+      ) {
+        this.setOutlineColor(
+          this.rgbToNumber(
+            gdjs.rgbOrHexToRGBColor(newObjectData.content.outlineColor)
+          )
+        );
+      }
+      if (oldObjectData.fillOpacity !== newObjectData.fillOpacity) {
+        this.setFillOpacity(newObjectData.fillOpacity);
+      }
+      if (oldObjectData.outlineOpacity !== newObjectData.outlineOpacity) {
+        this.setOutlineOpacity(newObjectData.outlineOpacity);
+      }
+      if (oldObjectData.outlineSize !== newObjectData.outlineSize) {
+        this.setOutlineSize(newObjectData.outlineSize);
+      }
+      return true;
+    }
+
+    /**
+     * Initialize the extra parameters that could be set for an instance.
+     */
+    extraInitializationFromInitialInstance(initialInstanceData): void {
+      if (initialInstanceData.customSize) {
+        this.setWidth(initialInstanceData.width);
+        this.setHeight(initialInstanceData.height);
+      }
+    }
+
+    private _updateTileMap(): void {
+      this._tileMapManager.getOrLoadTileMap(
+        this._tilemapJsonFile,
+        this._tilesetJsonFile,
+        (tileMap: TileMapHelper.EditableTileMap | null) => {
+          if (!tileMap) {
+            // getOrLoadTileMap already log errors.
+            return;
+          }
+
+          this._collisionTileMap = new gdjs.TileMap.TransformedCollisionTileMap(
+            tileMap,
+            this._collisionMaskTag
+          );
+          // The tile map polygons always keep the same references.
+          // It works because the tilemap is never modified.
+          this.hitBoxes = Array.from(
+            this._collisionTileMap.getAllHitboxes(this._collisionMaskTag)
+          );
+        }
+      );
+    }
+
+    updateHitBoxes(): void {
+      this.updateTransformation();
+      // Update the RuntimeObject hitboxes.
+      for (const hitboxes of this._collisionTileMap.getAllHitboxes(
+        this._collisionMaskTag
+      )) {
+        // RuntimeObject.hitBoxes point to the same polygons as the hitboxes
+        // from the grid. The hitboxes from the grid are updated according to
+        // the transformation at demand.
+        // This force all the hitboxes to update.
+        // The hitboxes array is built by _updateTileMap().
+      }
+      this.hitBoxesDirty = false;
+      this._renderer.redrawCollisionMask();
+      this.updateAABB();
+    }
```

Comment:
I'm not really proud of this.

Create a `updateAllHitboxes` in the model would need to add a method on each composition level which is not great and it's an operation that should not be seen as the "right way". It's needed only because `RuntimeObject` force to use an `Array`. So I think it's better to keep it outside of the model for clarity.

Looping here with the knowledge of the whole model is clearly not an option.

We could update the `Array` elements of `this.hitboxes`, it's not necessary as the polygon instances don't change, but it can make the code clearer.

Reply:
Could we do a method called "exceptionallyUpdateAllHitboxesEvenIfCostlyToCompute"? A long name while discourage usage and will be easy to find in the codebase should this be tried to be used again.